### PR TITLE
Use internal logging when context available

### LINKED ISSUE #4660: use internal logging package where appropriate
we have the package `internal/log` to execute hooks and give more debugging information when we log. specifically, it's useful because it tracks a unique ID for each request, and it makes tracking the lifecycle of logs easier

however, to do so, it requires us to pass the context everywhere. We haven't wired it all over the place, so many places use logrus instead of log.

with https://github.com/cri-o/cri-o/pull/4650 we're passing the context to more places, and as such we should add a follow up to use the log package in more places.

without looking too hard at the code, i'd say we should use the log package in *every* function that currently takes a context

### PR DESCRIPTION
**Pull request by ardaguclu**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

As stated in the issue, `internal/log` package stores more information than logrus and functions including context parameter should use `internal/log` package instead logrus.

#### Which issue(s) this PR fixes:
Fixes #4660 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by haircommander** [comment]
/ok-to-test

thanks for taking this on @ardaguclu !

I think it may benefit you to rebase on top of https://github.com/cri-o/cri-o/pull/4678, as I believe there are conflicts

File: cmd/crio/main.go [hunk 1]
```diff
@@ -63,22 +63,22 @@ func catchShutdown(ctx context.Context, cancel context.CancelFunc, gserver *grpc
 			case unix.SIGPIPE:
 				continue
 			case signals.Interrupt:
-				logrus.Debugf("Caught SIGINT")
+				log.Debugf(ctx, "Caught SIGINT")
```
Comment (haircommander) [inline cmd/crio/main.go:66]: @saschagrunert are the logging interceptors registered by this point? (I don't know how they work)

**Comment by fidencio** [comment]
LGTM, but needs a rebase, which I'd say to wait for @saschagrunert's feedback (most likely next week, due to the easter break) before doing it.

File: cmd/crio/main.go (hunk 1 above)
Comment (fidencio) [reply cmd/crio/main.go:66]: It looks like yes, the interceptors are registered by this point (and it seems like done a few lines before catchShutdown() is registered), but please, wait for @saschagrunert's input (as I'm probably wrong here ;-))

Comment (haircommander) [reply cmd/crio/main.go:66]: haha I believe you, I was being lazy and didn't check, so the fact you checked means it's there :+1: thanks :D

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests and a head nod from @saschagrunert

**Comment by haircommander** [comment]
CRI test failures are not the fault of this PR FYI

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by ardaguclu** [comment]
Is there anything I can do about failing tests like rebase, etc. Or retests eventually will work?

**Comment by saschagrunert** [comment]
> Is there anything I can do about failing tests like rebase, etc. Or retests eventually will work?

No, critest is currently broken.

/override critest_fedora
/override critest_rhel

But the crun integration should work as far as I can see.
/test integration_crun

**Comment by haircommander** [comment]
/override ci/openshift-jenkins/critest_fedora
/override ci/openshift-jenkins/critest_rhel

gh actions critest passed

File: internal/log/log_test.go [hunk 2]
```diff
@@ -137,4 +137,17 @@ var _ = t.Describe("Log", func() {
...
 		})
 	})
+
+	t.Describe("Fatalf", func() {
+		BeforeEach(func() { beforeEach(logrus.FatalLevel) })
+
+		It("should not error log", func() {
+			// Given
+			// When
+			log.Errorf(ctx, msg)
```
Comment (haircommander) [inline internal/log/log_test.go:147]: nit: I don't think this function does what we want it to

**Comment by haircommander** [comment]
/override ci/prow/e2e-gcp
(got e2e-agnositc)

**Pull request merged**